cognite/client/_api_client.py

```diff
@@ -328,7 +328,11 @@ def _retrieve_multiple(
         retrieved_items = tasks_summary.joined_results(lambda res: res.json()["items"])
 
         if identifiers.is_singleton():
-            return resource_cls._load(retrieved_items[0], cognite_client=self._cognite_client)
+            if retrieved_items:
+                return resource_cls._load(retrieved_items[0], cognite_client=self._cognite_client)
+            else:
+                # Data Models such as spaces to not raise a CogniteNotFoundError
```

Comment:
This comment was a little confusing to me, I suggest revising the formulation a little
```suggestion
                # Not all APIs (such as the Data Modeling API) return an error when unknown ids are provided, so we need to handle the unknown singleton identifier case here as well.
```